pkg/cloud/services/networking/securitygroups.go

```diff
@@ -147,21 +153,35 @@ func generateControlPlaneGroup(clusterName string) infrav1.SecurityGroup {
 					RemoteIPPrefix: "0.0.0.0/0",
 				},
 				{
-					Direction:      "ingress",
-					EtherType:      "IPv4",
-					PortRangeMin:   22,
-					PortRangeMax:   22,
-					Protocol:       "tcp",
-					RemoteIPPrefix: "0.0.0.0/0",
+					Direction:     "ingress",
+					EtherType:     "IPv4",
+					PortRangeMin:  2379,
+					PortRangeMax:  2380,
+					Protocol:      "tcp",
+					RemoteGroupID: remoteGroupIDSelf,
+				},
+				{
+					Direction:     "ingress",
+					EtherType:     "IPv4",
+					PortRangeMin:  10250,
+					PortRangeMax:  10250,
+					Protocol:      "tcp",
+					RemoteGroupID: remoteGroupIDSelf,
 				},
 			},
 			defaultRules...,
 		),
 	}
 }
 
-func generateGlobalGroup(clusterName string) infrav1.SecurityGroup {
-	secGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, globalSuffix)
+func (s *Service) generateWorkerGroup(clusterName string) infrav1.SecurityGroup {
+	secGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, workerSuffix)
+	secControlPlaneGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, controlPlaneSuffix)
+
+	secControlPlaneGroup, err := s.getSecurityGroupByName(secControlPlaneGroupName)
+	if err != nil || secControlPlaneGroup == nil || secControlPlaneGroup.ID == "" {
+		return infrav1.SecurityGroup{}
```

Comment:
so we have to have control plane secure group in order to create network ?
is there some edge case e.g controller doesn't use it while worker use it?
I knew it's rare so just ask in case we know it

Reply:
If ManagedSecurityGroups is true, control plane security group and worker security group will be created by following  [Check required ports](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#check-required-ports). If we do not like these rules, set ManagedSecurityGroups to false and specify existing security groups.

Reply:
thanks, the info works for me